lib/mappers/esdn_mapper.py

```diff
@@ -0,0 +1,331 @@
+from akara import logger
+from dplaingestion.utilities import iterify
+from dplaingestion.selector import exists, getprop
+from dplaingestion.mappers.oai_mods_mapper import OAIMODSMapper
+
+class EsdnMapper(OAIMODSMapper):
+    def __init__(self, provider_data):
+        super(EsdnMapper, self).__init__(provider_data)
+
+    def map_collection(self):
+        ret_dict = {"collection": getprop(self.provider_data, "collection")}
+
+        prop = self.root_key + "relatedItem"
+
+        if exists(self.provider_data, prop):
+            related_items = iterify(getprop(self.provider_data, prop))
+            host_types = [item for item in related_items if
+                          item.get("type") == "host" and item.get("displayLabel") == "Collection"]
+
+            for host_type in host_types:
+                title = getprop(host_type, "titleInfo/title", True)
+                if title:
+                    ret_dict["collection"]["title"] = title
+
+                description = getprop(host_type, "abstract", True)
+                if description:
+                    ret_dict["collection"]["description"] = description
+
+        if ret_dict["collection"]:
+            self.update_source_resource(ret_dict)
+
+    # Copied from DigitalNC
+    def map_creator_and_contributor(self):
+        prop = self.root_key + "name"
+        _dict = {
+            "creator": [],
+            "contributor": []
+        }
+
+        if exists(self.provider_data, prop):
+            for s in iterify(getprop(self.provider_data, prop)):
+                name = s.get("namePart")
+                if name:
+                    role_terms = []
+                    try:
+                        for r in iterify(s.get("role")):
+                            if isinstance(r.get("roleTerm"), dict):
+                                role_terms.append(r.get("roleTerm").get("#text").lower())
+                            else:
+                                role_terms.append(r.get("roleTerm").lower())
```

Comment:
From lines 47 to here, the only possibilities considered are that the `roleTerm` property is dict or string. Are we certain it could not be an array?